internal/tally/tally.go

```diff
@@ -16,27 +15,18 @@ type (
 		fx.In
 		Lifecycle fx.Lifecycle
 		Config    *config.Config
-		Reporter  tally.StatsReporter `optional:"true"`
+		Reporter  tally.StatsReporter
 	}
 )
 
-const (
-	reportingInterval = time.Second
-)
-
 func NewRootScope(params MetricParams) tally.Scope {
-	// XXX: Inject your own reporter here.
-	reporter := params.Reporter
```

Comment:
why make this change?  if params.Reporter is null wouldn't this cause a panic in the code below.  If we don't set the default, then at a minimum we'll have to make sure it's set in all code paths, including tests.

Reply:
Forget to add the stats reporter file. It will always be non null. If statsd is not configured , it will provide non null tallynull reporter